src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+    let value = [];
```

Comment:
It is not necessary to bring the definition of a variable that is used in the middle of the loop outside.
You should also think about better naming.